pkg/utils/kubeconfig/kubeconfig.go

```diff
@@ -189,6 +189,18 @@ func AppendAuthenticator(config *clientcmdapi.Config, clusterMeta *api.ClusterMe
 			args = append(args, "--region", clusterMeta.Region)
 		}
 	}
+	// If the alpha API version is selected, check the kubectl version
+	// If kubectl 1.24.0 or above is detected, override with the beta API version
+	// kubectl 1.24.0 removes the alpha API version, so it will never work
+	// Therefore as a best effort try the beta version even if it might not work
+	if execConfig.APIVersion == alphaAPIVersion {
+		kubectlVersion := getKubectlVersion()
+		// Silently ignore errors because kubectl is not required to run eksctl
+		compareVersions, err := utils.CompareVersions(kubectlVersion, "1.24.0")
+		if kubectlVersion != "" && err == nil && compareVersions >= 0 {
+			execConfig.APIVersion = betaAPIVersion
+		}
```

Comment:
```suggestion
		if kubectlVersion := getKubectlVersion(); kubectlVersion != "" {
			// Silently ignore errors because kubectl is not required to run eksctl
			compareVersions, err := utils.CompareVersions(kubectlVersion, "1.24.0")
			if err == nil && compareVersions >= 0 {
				execConfig.APIVersion = betaAPIVersion
			}
		}
```